### PR TITLE
ipadnszone: Accept localhost and localnet in allow_query.

### PR DESCRIPTION
**Pull request by zpericic**
Bind allows localhost and localnet in allow query so we should accept it.

### CODE REVIEW

**Comment by rjeffman** [comment]
As FreeIPA manages users and hosts in a central location, what is the use case to have its integrated DNS nameserver to only respond to local queries?

**Comment by zpericic** [comment]
> As FreeIPA manages users and hosts in a central location, what is the use case to have its integrated DNS nameserver to only respond to local queries for a specific zone?

It is accepted by bind, bind-dyndb-ldap and IPA so why wouldn't we allow it... 

I use it in conjunction with other ACIs eg. "localhost; 10/8;". 

There is many use cases when you want to be able resolve domain on DNS machine. It's not disallowed to use other software on IPA server. It's recommended not to use any network service which could compromise IPA, but it's not mandatory. and you may want to resolve from some local software.

Some examples are null client for mail, remote logging, monitoring...

File: plugins/modules/ipadnszone.py [hunk 1]
```diff
@@ -257,6 +257,8 @@ def validate_ips(self, ips, error_msg):
                 is_ip_address(ip),
                 is_ip_network_address(ip),
                 ip == "any",
+                ip == "localhost",
+                ip == "localnets",
                 ip == "none"
```
Comment (rjeffman) [inline plugins/modules/ipadnszone.py:262]: Please, change this to `ip in ["any", "localhost", "localnets", "none"]`.

**Review by rjeffman — CHANGES_REQUESTED** [review]

**Review by rjeffman — APPROVED** [review]
LGTM.

**Pull request merged**